src/blog/2024/05/flowfuse-on-robustel-eg5120.md

```diff
@@ -0,0 +1,93 @@
+    - robustel
+    - eg5120
+---
+
+The [robustel EG5120](https://www.robustel.com/product/eg5120-industrial-edge-computing-gateway/) is a versatile gateway that facilitates robust connectivity for industrial IoT applications. Integrating this powerful hardware with FlowFuse not only enhances its capabilities but also simplifies the management and deployment process. In this guide, we’ll walk through the steps to integrate the robustel EG5120 with FlowFuse, making your IoT solutions more efficient and easier to manage.
+
+<!--more-->
+
+#### **Introduction**
+
+This blog is designed for IoT enthusiasts and professionals who are looking to enhance their industrial IoT applications. The [robustel EG5120](https://www.robustel.com/product/eg5120-industrial-edge-computing-gateway/), equipped with Linux-based Debian 11 and preinstalled Node-RED, offers robust connectivity options. When combined with FlowFuse, this gateway becomes even more powerful, enabling seamless device management and deployment. 
+
+
+#### **Robustel and Node-RED**
+
+The robustel EG5120 supports multiple connectivity options including Ethernet, Wi-Fi, and cellular networks, which are essential for flexible deployments in various industrial scenarios. Its built-in support for MQTT and Modbus protocols facilitates seamless integration with a wide array of IoT devices and services. This blog will guide you through using FlowFuse to effectively manage your Node-RED instance, enhancing both the security and scalability of your IoT applications.
+
+#### **Migration**
+
+If you're looking to migrate your Node-RED instance, follow the steps in [this migration guide](/docs/migration/node-red-tools/) to back up and transfer your configurations safely. By incorporating the FlowFuse device agent, we aim to replace the preinstalled Node-RED runtime, thus providing enhanced monitoring, automated backups, and improved management capabilities. This transition will make your deployments more secure and resilient.
+
+#### **Setting Up the Environment**
+
+**Important Notice:** The device encounters an issue where the SuperUser and the default user are running different versions of Node.js. The SuperUser is running an outdated version incompatible with our needs. For the FlowFuse device agent, a minimum of NodeJS v14 is required, though v20 or later is recommended. For the latest recommendations, check this [documentation](/docs/device-agent/install/#prerequisites).
+
+**1. Gaining Root Access:**
+   - Start by opening your terminal and gain root access to adjust system configurations crucial for the following steps:
+     ```bash
+     sudo su
+     ```
+   - Verify the version of Node.js:
+     ```bash
+     node -v
+     ```
+   - If the version is below v14, proceed to the next step to upgrade. If it's v14 or higher, you might opt to skip to step 4.
+
+**2. Updating Node.js:**
+   - To ensure optimal functionality, update or install Node.js v20.12 using the Node version manager:
```

Comment:
Is there a way to install the latest stable version? That makes the post less temporal.

Reply:
The version of node.js that is v20.12 is LTS.  I can document that.